lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart

```diff
@@ -0,0 +1,308 @@
+            break;
+          }
+        }
+      } else if (emotePacks[packSearch] != null) {
+        for (final emote in emotePacks[packSearch]!.images.entries) {
+          if (emote.key.toLowerCase().contains(emoteSearch)) {
+            ret.add({
+              'type': 'emote',
+              'name': emote.key,
+              'pack': packSearch,
+              'pack_avatar_url': emotePacks[packSearch]!.pack.avatarUrl?.toString(),
+              'pack_display_name': emotePacks[packSearch]!.pack.displayName ?? packSearch,
+              'mxc': emote.value.url.toString(),
+            });
+          }
+          if (ret.length > maxResults) {
+            break;
+          }
+        }
+      }
+      // TODO:
+      // aside of emote packs, also propose normal (tm) unicode emojis
+      // final matchingUnicodeEmojis = Emoji.all()
+      // .where(
+      //   (element) => [element.name, ...element.keywords]
+      //       .any((element) => element.toLowerCase().contains(emoteSearch)),
+      // )
+      // .toList();
+      final matchingUnicodeEmojis = [];
+      // sort by the index of the search term in the name in order to have
+      // best matches first
+      // (thanks for the hint by github.com/nextcloud/circles devs)
+      matchingUnicodeEmojis.sort((a, b) {
+        final indexA = a.name.indexOf(emoteSearch);
+        final indexB = b.name.indexOf(emoteSearch);
+        if (indexA == -1 || indexB == -1) {
+          if (indexA == indexB) return 0;
+          if (indexA == -1) {
+            return 1;
+          } else {
+            return 0;
+          }
+        }
+        return indexA.compareTo(indexB);
+      });
+      for (final emoji in matchingUnicodeEmojis) {
+        ret.add({
+          'type': 'emoji',
+          'emoji': emoji.char,
+          // don't include sub-group names, splitting at `:` hence
+          'label': '${emoji.char} - ${emoji.name.split(':').first}',
+          'current_word': ':$emoteSearch',
+        });
+        if (ret.length > maxResults) {
+          break;
+        }
+      }
+    }
+    final userMatch = RegExp(r'(?:\s|^)@([-\w]+)$').firstMatch(searchText);
+    if (userMatch != null) {
+      final userSearch = userMatch[1]!.toLowerCase();
+      for (final user in room.getParticipants()) {
+        if ((user.displayName != null &&
+                (user.displayName!.toLowerCase().contains(userSearch) ||
+                    slugify(user.displayName!.toLowerCase()).contains(userSearch))) ||
+            user.id.split(':')[0].toLowerCase().contains(userSearch)) {
+          ret.add({
+            'type': 'user',
+            'mxid': user.id,
+            'mention': user.mention,
+            'displayname': user.displayName,
+            'avatar_url': user.avatarUrl?.toString(),
+          });
+        }
+        if (ret.length > maxResults) {
+          break;
+        }
+      }
+    }
+    final roomMatch = RegExp(r'(?:\s|^)#([-\w]+)$').firstMatch(searchText);
```

Comment:
Recommend define this regex pattern somewhere , like pattern constants 

And we define good naming regex there

Example: mention_regex, naming_regex ...etc..